Spigot-API-Patches/0039-LootTable-API.patch

```diff
@@ -0,0 +1,303 @@
++
++import org.bukkit.entity.Entity;
++
++public interface LootableEntityInventory extends LootableInventory {
++
++    /**
++     * Gets the entity that is lootable
++     * @return The Entity
++     */
++    Entity getEntity();
++}
+diff --git a/src/main/java/com/destroystokyo/paper/loottable/LootableInventory.java b/src/main/java/com/destroystokyo/paper/loottable/LootableInventory.java
+new file mode 100644
+index 0000000..eaa01cf
+--- /dev/null
++++ b/src/main/java/com/destroystokyo/paper/loottable/LootableInventory.java
+@@ -0,0 +1,94 @@
++package com.destroystokyo.paper.loottable;
++
++import org.bukkit.entity.Player;
++
++import java.util.UUID;
++
++public interface LootableInventory extends Lootable {
++
++    /**
++     * Whether or not this object has ever been filled
++     * @return Has ever been filled
++     */
++    boolean hasBeenFilled();
++
++    /**
++     * Has this player ever looted this block
++     * @param player The player to check
++     * @return Whether or not this player has looted this block
++     */
++    default boolean hasPlayerLooted(Player player) {
++        return hasPlayerLooted(player.getUniqueId());
++    }
++
++    /**
++     * Has this player ever looted this block
++     * @param player The player to check
++     * @return Whether or not this player has looted this block
++     */
++    boolean hasPlayerLooted(UUID player);
++
++    /**
++     * Gets the timestamp, in milliseconds, of when the player last looted this object
++     *
++     * @param player The player to check
++     * @return Timestamp last looted, or null if player has not looted this object
++     */
++    default Long getLastLooted(Player player) {
++        return getLastLooted(player.getUniqueId());
++    }
++
++    /**
++     * Gets the timestamp, in milliseconds, of when the player last looted this object
++     *
++     * @param player The player to check
++     * @return Timestamp last looted, or null if player has not looted this object
++     */
++    Long getLastLooted(UUID player);
++
++    /**
++     * Change the state of whether or not a player has looted this block
++     * @param player The player to change state for
++     * @param looted true to add player to looted list, false to remove
++     * @return The previous state of whether the player had looted this or not
++     */
++    default boolean setHasPlayerLooted(Player player, boolean looted) {
++        return setHasPlayerLooted(player.getUniqueId(), looted);
++    }
++
++    /**
++     * Change the state of whether or not a player has looted this block
++     * @param player The player to change state for
++     * @param looted true to add player to looted list, false to remove
++     * @return The previous state of whether the player had looted this or not
++     */
++    boolean setHasPlayerLooted(UUID player, boolean looted);
++
++    /**
++     * Returns Whether or not this server has auto refill enabled for this world
```

Comment:
Odd that the description is saying "enabled for this world" but we're talking about a `LootableInventory`, not a `World`. Is it per world or per `LootableInventory`?